providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
Ok, there is no way to properly support _ALL_ regexes. Some examples from fcontext -l:
* /var/lib/amanda/[^/]+/index(/.*)?
* /var/mailman/pythonlib(/.*)?/.+\.so(\..*)?
* /usr/lib(64)?/vmware/lib(/.*)?/HConfig\.so

So I came up with a simple and transparent solution. Tell me what you think about it and if it is ok, I will instantly implement it and send you a pull request.
```ruby
def get_restorecon_path (regex_path)
  path = regex.path.sub(/\\/,'') # Remove backslashes
  return path if File.exist?(path) # If regular file, just return
  path.count('/').times do
    path = File.dirname(path) # Splits at last '/' and returns front part
    return path if File.directory?(path) # Return is path is a valid directory
  end
  return '/' # This actually should never happen.
end
```
In the end we will just use restorecon -R #{get_restorecon_path(new_resource.file_spec)}

What do you think...